k3s-1.26.yaml

```diff
@@ -0,0 +1,173 @@
+      mv dist/artifacts/k3s* dist/artifacts/embedded-k3s
+  # Next build things using wolfi components.
+  - uses: patch
+    with:
+      patches: 0001-PATCH-Wolfi-support.patch
+  - runs: |
+      go mod tidy
+
+      ./scripts/build
+
+      # Move the "inner" k3s binary out of the bundle dir and into some place permanent
+      mkdir -p "${{targets.destdir}}"/bin/
+      mv bin/k3s "${{targets.destdir}}"/bin/_k3s
+
+      # This time around, we use a mostly empty bin/ dir with only symlinks to the "inner" k3s binary
+      ./scripts/package-cli
+
+      # Finally, install the "outer" k3s multicall binary
+      install -Dm755 dist/artifacts/k3s* "${{targets.destdir}}"/bin/k3s
+  - uses: strip
+
+subpackages:
+  - name: k3s-embedded-1.26
+    description: "k3s with embedded components"
+    dependencies:
+      runtime:
+        - busybox
+        - ip6tables
+        - libseccomp
+        - conntrack-tools
+      provides:
+        - k3s-embedded=1.26
+    pipeline:
+      - runs: |
+          # Install the "outer" k3s multicall binary into the package path
+          install -Dm755 dist/artifacts/embedded-k3s "${{targets.subpkgdir}}"/bin/k3s
+
+  - name: k3s-images-1.26
+    description: "k3s images, pre-bundled"
+    dependencies:
+      provides:
+        - k3s-images=1.26
+    pipeline:
+      - runs: |
+          set -e -u -x
+
+          _arch=$(go env GOARCH)
+
+          images=""
+          while read -r line
+          do
+            repo=$(echo $line | cut -d ':' -f 1)
+            tag=$(echo $line | cut -d ':' -f 2)
+
+            # Replace upstream images with cgr.dev equivalents only if they exist
+            case $repo in
+              # docker.io/rancher/klipper-helm)
+              #   image=$(crane digest --full-ref cgr.dev/chainguard/klipper-helm:latest)
+              #   ;;
+              # docker.io/rancher/klipper-lb)
+              #   image=$(crane digest --full-ref cgr.dev/chainguard/klipper-lb:latest)
+              #   ;;
+              # docker.io/rancher/local-path-provisioner)
+              #   image=$(crane digest --full-ref cgr.dev/chainguard/local-path-provisioner:latest)
+              #   ;;
+              docker.io/rancher/mirrored-metrics-server)
+                image=$(crane digest --full-ref cgr.dev/chainguard/metrics-server:latest)
+                ;;
+              docker.io/rancher/mirrored-coredns-coredns)
+                image=$(crane digest --full-ref cgr.dev/chainguard/coredns:latest)
+                ;;
+              docker.io/rancher/mirrored-library-busybox)
+                image=$(crane digest --full-ref cgr.dev/chainguard/busybox:latest)
+                ;;
+              docker.io/rancher/mirrored-library-traefik)
+                image=$(crane digest --full-ref cgr.dev/chainguard/traefik:latest)
+                ;;
+              *)
+                image="$repo:$tag@$(crane digest $repo:$tag)"
```

Comment:
I'm wondering if we should fail on unknown do signal that we have gaps?

Reply:
these haven't changed in years, but I like that idea